### PR TITLE
VDA-eMerge-Underlay-RC1

### PR DESCRIPTION
**Pull request by chenchals**
- Added underlay for eMerge Record Counter 
- deployed app to local docker and tested

### CODE REVIEW

**Review by marikomedlock — COMMENTED** [review]
So exciting to see this progress on the eMerge underlay!

I left some comments inline, but it's fine with me to merge this first pass as is, and tackle those in follow-on PRs.

File: service/src/main/resources/application.yml [hunk 1]
```diff
@@ -48,7 +48,8 @@ tanagra:
         type: REGRESSION_TEST
 
 logging.pattern.level: '%X{requestId} %5p'
-logging.level.bio.terra.tanagra: DEBUG
+#logging.level.bio.terra.tanagra: DEBUG
```
[the remark this answers is not in the document]
Comment (chenchals) [reply service/src/main/resources/application.yml:51]: nope. rolled bacl

File: underlay/src/main/resources/config/criteria/emerge/criteriaselector/site/selector.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "tanagra-sitee",
```
[the remark this answers is not in the document]
Comment (chenchals) [reply underlay/src/main/resources/config/criteria/emerge/criteriaselector/site/selector.json:2]: :-)

File: underlay/src/main/resources/config/datamapping/emerge/entity/height/all.sql [hunk 3]
```diff
@@ -0,0 +1,6 @@
+SELECT
+    ROW_NUMBER() OVER() AS row_id,
+    bo.IND_SEQ as person_id,
+    bo.height AS value_as_number,
+    bo.age_at_event AS age_at_occurrence
+FROM `${omopDataset}.bmi` bo
```
[the remark this answers is not in the document]
Comment (chenchals) [reply underlay/src/main/resources/config/datamapping/emerge/entity/height/all.sql:6]: no, in this case the bmi-table has both height and weight

File: underlay/src/main/resources/config/datamapping/sd/entity/phenotype/childParent.sql [hunk 4]
```diff
@@ -0,0 +1,3 @@
+SELECT criteria_meta_seq AS child, parent_seq AS parent
+FROM `${omopDataset}.phewas_criteria`
```
[the remark this answers is not in the document]
Comment (chenchals) [reply underlay/src/main/resources/config/datamapping/sd/entity/phenotype/childParent.sql:2]: not same as phewas

File: underlay/src/main/resources/config/datamapping/sd/entity/phenotype/all.sql [hunk 5]
```diff
@@ -0,0 +1,8 @@
+SELECT
+    criteria_meta_seq AS id,
+    type,
+    REGEXP_EXTRACT(label, r'^PHEWAS_([0-9.]+)-') AS concept_code,
+    REGEXP_EXTRACT(label, r'^PHEWAS_[0-9.]+-(.*)') AS name,
+    REGEXP_EXTRACT(label, r'^PHEWAS_(.*)') AS label,
+    PARSE_NUMERIC(REGEXP_EXTRACT(label, r'^PHEWAS_([0-9.]+)-')) AS numeric_code
+FROM `${omopDataset}.phewas_criteria`
```
[the remark this answers is not in the document]
Comment (chenchals) [reply underlay/src/main/resources/config/datamapping/sd/entity/phenotype/all.sql:8]: similar but not same :-(

File: underlay/src/main/resources/config/datamapping/emerge/entity/weight/all.sql [hunk 6]
```diff
@@ -0,0 +1,6 @@
+SELECT
+    ROW_NUMBER() OVER() AS row_id,
+    bo.IND_SEQ as person_id,
+    bo.weight AS value_as_number,
+    bo.age_at_event AS age_at_occurrence
+FROM `${omopDataset}.bmi` bo
```
[the remark this answers is not in the document]
Comment (chenchals) [reply underlay/src/main/resources/config/datamapping/emerge/entity/weight/all.sql:6]: no, in this case the bmi-table has both height and weight

File: underlay/src/main/resources/config/datamapping/emerge/entity/icd10cmOccurrence/all.sql [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+JOIN (
+    SELECT
+        pc.criteria_meta_seq as concept_id,
+        regexp_extract(pc.name, '.*-(.*)') as name,
+        regexp_extract(pc.name, '(.*)-.*') as concept_code,
+        pc.is_leaf
+    FROM `${omopDataset}.icd10_criteria` pc
+    WHERE pc.parent_seq < (
+        select criteria_meta_seq from `${omopDataset}.icd10_criteria`
+        where starts_with(label, 'ICD10PCS')
```
[the remark this answers is not in the document]
Comment (chenchals) [reply underlay/src/main/resources/config/datamapping/emerge/entity/icd10cmOccurrence/all.sql:18]: No this is correct - for ICD10CM- Deployed and checked this to be correct

File: underlay/src/main/resources/config/datamapping/emerge/entity/icd10cm/childParent.sql [hunk 8]
```diff
@@ -0,0 +1,9 @@
+SELECT
+    pc.criteria_meta_seq AS child,
+    pc.parent_seq AS parent
+FROM  `${omopDataset}.icd10_criteria` pc
+WHERE pc.parent_seq < (
+    select criteria_meta_seq from `${omopDataset}.icd10_criteria`
+    where starts_with(label, 'ICD10PCS')
```
[the remark this answers is not in the document]
Comment (chenchals) [reply underlay/src/main/resources/config/datamapping/emerge/entity/icd10cm/childParent.sql:7]: No this is correct - for ICD10CM- Deployed and checked this to be correct

**Review by freemabd — APPROVED** [review]
LGTM

**Pull request merged**